### PR TITLE
Fix Scheduler IntelliJ warnings

### PR DESCRIPTION
**Pull request by dain**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 1]
```diff
@@ -176,17 +170,10 @@ public Node selectCurrentNode()
         {
             checkArgument(limit > 0, "limit must be at least 1");
 
-            final String coordinatorIdentifier = nodeManager.getCurrentNode().getNodeIdentifier();
+            String coordinatorIdentifier = nodeManager.getCurrentNode().getNodeIdentifier();
```
Comment (nileema) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: You can use ``nodeManger.getCoordinators() `` to get all the coordinators

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: Filed #2868

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 2]
```diff
@@ -195,12 +182,11 @@ public boolean apply(Node node)
...
 
         /**
-         * Identifies the nodes for runnning the specified splits.
+         * Identifies the nodes for running the specified splits.
          *
          * @param splits the splits that need to be assigned to nodes
-         *
          * @return a multimap from node to splits only for splits for which we could identify a node to schedule on.
-         *      If we cannot find an assignment for a split, it is not included in the map.
+         * If we cannot find an assignment for a split, it is not included in the map.
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: I think this should be aligned to the `@return`. That's how it is shown in the "How to Write Doc Comments for the Javadoc Tool". Though it might be better to move this sentence to the description (and remove the period for `@return`).

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 3]
```diff
@@ -231,11 +217,10 @@ public boolean apply(Node node)
                 checkCondition(!candidateNodes.isEmpty(), NO_NODES_AVAILABLE, "No nodes available to run query");
 
                 // compute and cache number of splits currently assigned to each node
-                for (Node node : candidateNodes) {
-                    if (!splitCountByNode.containsKey(node)) {
-                        splitCountByNode.put(node, nodeTaskMap.getPartitionedSplitsOnNode(node));
-                    }
-                }
+                candidateNodes.stream()
+                        .filter(node -> !splitCountByNode.containsKey(node))
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: They should have made these `include/exclude`

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 4]
```diff
@@ -298,13 +280,10 @@ public boolean apply(Node node)
...
                 if (!hint.hasPort() || split.isRemotelyAccessible()) {
-                    for (Node node : nodeMap.getNodesByHost().get(address)) {
-                        if (includeCoordinator || !coordinatorIdentifier.equals(node.getNodeIdentifier())) {
-                            if (chosen.add(node)) {
-                                scheduleLocal.incrementAndGet();
-                            }
-                        }
-                    }
+                    nodeMap.getNodesByHost().get(address).stream()
+                            .filter(node -> includeCoordinator || !coordinatorIdentifier.equals(node.getNodeIdentifier()))
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: This has the same logic as the one above. Too bad you can't save and apply the same stream logic to both.

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java [hunk 5]
```diff
@@ -355,30 +334,29 @@ public boolean apply(Node node)
...
+                HostAddress coordinatorHostAddress = nodeManager.getCurrentNode().getHostAndPort();
+                if (split.getAddresses().stream().anyMatch(host -> canSplitRunOnHost(split, coordinatorHostAddress, host))) {
+                    chosen.add(nodeManager.getCurrentNode());
                 }
             }
 
             return ImmutableList.copyOf(chosen);
         }
+
+        private boolean canSplitRunOnHost(Split split, HostAddress coordinatorAddress, HostAddress host)
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: Name `coordinatorHost` to match `host`

File: presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java [hunk 6]
```diff
@@ -846,11 +843,12 @@ private boolean addNewExchangesAndBuffers()
     }
 
     @VisibleForTesting
+    @SuppressWarnings("NakedNotify")
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java]: Why this warning? The `this.notifyAll()` call is inside a `synchronized (this)` block.

File: presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java (hunk 3 above)
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java]: what?

File: presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java (hunk 6 above)
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java]: `doUpdateState` is invoked from a callback on the state machine object when the state changes

Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java]: Also, I don't think this commit should not change behavior, just modify the code to match style and rules.  We could do it in another commit, but I am rewriting this whole class.
